packages/block-editor/src/components/iframe/use-scale-canvas.js

```diff
@@ -145,12 +167,17 @@ export function useScaleCanvas( {
 	maxContainerWidth = 750,
 	scale,
 } ) {
-	const [ contentResizeListener, { height: contentHeight } ] =
-		useResizeObserver();
+	const [ { height: contentHeight }, setContentRect ] = useState( NULL_SIZE );
+	const contentRefCallback = useResizeObserver( ( entries ) => {
+		setContentRect( extractSize( entries ) );
```

Comment:
The legacy implementation is slightly more efficient because it sets state and triggers a rerender only if one of the sizes (height, width) really changes.

I don't know if this matters in practice. Is there a scenario where we're getting a series of `ResizeObserver` events that all report the same size?

One way to fix this is to have two separate `useState`s for `height` and `width`.

Or make use of the fact that the hooks uses only the `height` and have state only for `height`.

Reply:
> The legacy implementation is slightly more efficient because it sets state and triggers a rerender only if one of the sizes (height, width) really changes.

I'd prefer to keep the code simpler for now. If we run into any noticeable performance issues around it here or elsewhere, then maybe it's worth making a wrapper on `useResizeObserver` that everyone can use that incorporates the performance improvements.

> Is there a scenario where we're getting a series of `ResizeObserver` events that all report the same size?

I'm not seeing that happen at all in my testing.

The `contentHeight` technically doesn't need to be updated all the time, but we're already getting a re-render from the other `useResizeObserver`, so I don't think it matters in practice here.